src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy

```diff
@@ -215,10 +241,21 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
       void afterResolve(ResolvableDependencies resolvableDependencies) {}
     })
 
-    project.tasks.withType(AbstractCompile).all { AbstractCompile compile ->
-      CheckerFrameworkTaskExtension checkerExtension =
-          compile.extensions.create("checkerFramework", CheckerFrameworkTaskExtension)
-    }
+    configureTasks(project, AbstractCompile, { AbstractCompile compile ->
+      def ext = compile.extensions.findByName("checkerFramework")
+      if (ext == null) {
+        LOG.info("Adding checkerFramework extension to task {}", compile.name)
+        compile.extensions.create("checkerFramework", CheckerFrameworkTaskExtension)
+      } else if (ext instanceof CheckerFrameworkTaskExtension) {
+        LOG.
+            warn("Task {} in project {} already has checkerFramework added to it;" +
+                " make sure you're applying the org.checkerframework plugin after the Java plugin", compile.name,
```

Comment:
Frankly speaking, I don't think this deserves `warn` level. It could be `debug`.

`extension already exists` means `checkerframework code` is executed multiple times, which is `checkerframework's` internal details (issues) rather that the user did something wrong.